lib/rbac/access.rb

```diff
@@ -25,22 +25,40 @@ def accessible?
     end
 
     def id_list
-      ids = collect_ids
-      Rails.logger.info("IDS for #{@app_name} #{ids}")
-      ids.include?('*') ? [] : ids
+      ids
+      Rails.logger.info("IDS for #{@app_name} #{@ids}")
+      @ids.include?('*') ? [] : @ids
+    end
+
+    def owner_scoped?
+      ids
+      @ids.include?('*') ? false : owner_scope_filter?
     end
 
     def self.enabled?
       ENV['BYPASS_RBAC'].blank?
     end
 
     private
-    def collect_ids
-      @acl.collect do |item|
-        item.resource_definitions.collect do |rd|
-          rd.attribute_filter.value
+
+    def ids
+      @ids ||= @acl.each_with_object([]) do |item, ids|
+        item.resource_definitions.each do |rd|
+          next unless rd.attribute_filter.key == 'id'
+          next unless rd.attribute_filter.operator == 'equal'
+          ids << rd.attribute_filter.value
+        end
+      end
+    end
+
+    def owner_scope_filter?
+      @acl.any? do |item|
+        item.resource_definitions.any? do |rd|
+          rd.attribute_filter.key == 'owner' &&
```

Comment:
For my understanding of how RBAC sees these attribute_filters.  The key 'owner' does that mean the owner of the resource_definition?  e.g. the person who created it? and/or has write access to it?

Reply:
@syncrou Its the attribute name for the resource you want to set the permissions for. The attribute filter comprises of 3 pieces

1. Attribute name (key)
2. Operator (operator)
3. Value       (value)

For the owner scope we set it to be
key = "owner"
operator = "equal"
value = "{{username}}"